lib/saml_idp/request.rb

```diff
@@ -115,10 +115,11 @@ def valid_signature?
     end
 
     def service_provider?
-      service_provider.valid?
+      service_provider && service_provider.valid?
```

Comment:
@sbc100 you are right, I did claim that about `fingerprint` and `metadata_url` but I think I was wrong. I was reading the specs but not closely enough.

Here's the current behavior:

```
sp = ServiceProvider.new({})  # invalid
sp = ServiceProvider.new({foo: bar})  # valid
sp = ServiceProvider.new({identifier: nil})  # valid
sp = ServiceProvider.new({identifier: 'some/thing'})  # valid
```

There's nothing about the ServiceProvider class as implemented, all by itself, that requires a not-nil `identifier` value. You're saying that it should?